### DISCUSSION TITLE
Adding own property to DisplayObjects

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7711] [Q&A]
There is "name" property already, but I want to add "index" (number), too
Because "index" is useful if there are multiple same "name" instances to check which one is clicked exactly.
Extending Container or Sprite as class is not enough, Because event.target is DisplayObject,
So I can't use Typescript type checking.
Is there any way to add all DisplayObjects my own property in Typescript way?

**Comment by ivanpopelyshev** [comment]
yes, in global.d.ts how its done in cacheAsBitmap package. You cloned pixi, right? just add your own global.d.ts somewhere and triple-slash-reference it from any of your ts files

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129630943-c785a810-f24c-4f60-ad8c-8ff19ce58d8f.png)

I added here. Is it ok?

**Comment by owendwyer** [comment]
> Extending Container or Sprite as class is not enough, Because event.target is DisplayObject,

Another solution would be to cast event.target to your class

let myClickedObject = event.target as MyContainerClass;
console.log(myClickedObject.index);
